ansible_image_validation/validation-playbooks/repo-validation-check.yaml

```diff
@@ -0,0 +1,128 @@
+    line: "\n {{ IncorrectPackageAccToArch }}"
+    create: yes
+    state: present 
+  when: ( offer_type!='byol') and ("{{ IncorrectPackageAccToArch }}" in output_package_acc_architecture.stdout_lines)  
+
+
+
+- name: Check if the client package is present according to the offertype when offertype is not equal to byol.
+  ansible.builtin.shell:
+    cmd: |
+      #!/bin/bash
+          client_package=$( rpm -qa | grep rhui )                                                                           ## Value of client package is getting calculated
+          if [[ "{{ architecture }}" -eq "arm64" ]]; then                              
+              substring="arm64-"                                                       
+              clientpkgwithoutarchitecture="${client_package//$substring/}"                                                 ## The arm64 substing is removed so that common validation logic can be used for the client packages of both the types of architectures.
+              client_package=$clientpkgwithoutarchitecture                            
+          fi                            
+                                                          
+                      
+              package_type=$(echo $client_package | sed 's/-[0-9]\+\.[0-9]\+-[0-9]\+\.noarch//')                            ## Eliminate the suffix  "-2.3-5343.noarch.rpm"  from client package. Remaining value should be equal to "rhui-azure-rhel8" + "offertype" | Examples: "rhui-azure-rhel8", "rhui-azure-rhel8-eus", "rhui-azure-rhel8-sap-ha"
+              prefix=$(echo $client_package | sed -E 's/^([^-]+-[^-]+-[^-]+)-.*/\1/')                                       ## Extract prefix rhui-azure-rhel8"  value from client package
+              extract_offer_val=$(echo $package_type | sed "s/$prefix//")                                                   ## For Base Offertype, extract_offer_val should be equal to null.
+
+
+              offertype={{ offer_type }}
+              offertype=${offertype//-/}  
+              extract_offer_val=${extract_offer_val//-/}
+
+              echo "Print the value of the offertype passed in the pipeline variables: " $offertype
+              echo "Print the value of the offertype which is present the client package name: " $extract_offer_val
+  
+              if [[  $offertype  == "base" ]]; then
+                  if [[ -z "$extract_offer_val" ]]; then 
+                        echo "{{ CorrectPackageAccToOffer }}"
+                  else
+                        echo "{{ IncorrectPackageAccToOffer }}"                                                             ## This if block refers to the check for the base offertype which is supplied through pipeline variables. The value of extract_offer_val variable should be null. Example of Base Offertype client package "rhui-azure-rhel9-2.3-655.noarch". It does not contain the value of any offertype of RHEL Images.
+                  fi   
+
+              else    
```

Comment:
![image](https://github.com/user-attachments/assets/45244a84-c7fc-40b5-b9cb-15d759e84c75)

^^these are the 'offers' as per image pipeline

![image](https://github.com/user-attachments/assets/84955abd-de5c-4862-bd95-27d3f919eaf8)


^^These are client package names. 

I notice that discrepancy in 'sap-apps' expected by pipeline but 'sapapps' expected by offer in client package. 
Can you handle that?

Reply:
Please also confirm if there are any more such discrepancies